src/Messaging.ts

```diff
@@ -103,13 +103,25 @@ export class Messaging {
       id: typeof id === 'string' ? id : utils.convertToHexString(id),
       subject
     }
-    await this.provider.postToEndpoint(`messages/${counterPartyAddress}`, {
-      type,
-      message: JSON.stringify(message)
-    })
+    await this.sendMessage(counterPartyAddress, type, message)
     return message
   }
 
+  public async paymentMessage(
+    counterPartyAddress: string,
+    messageId: string,
+    message: string
+  ): Promise<PaymentMessage> {
+    const type = 'PaymentMessage'
+    const data = {
```

Comment:
That's not possible, there's already something called message

Reply:
I could rename `message` into `paymentMessage` to rename `data` to message`, but then then it would clash with the function name `paymentMessage. I could rename the function name to `sendPaymentMessage` but it would not be coherent with the other function in this class (it is `paymentRequest()` and not `sendPaymentRequest()`).

Reply:
True. Sorry I went to fast over it.
So I would argue the function parameter is named in a wrong way. The `message` is apparently the object you pass to the `sendMessage()` function. And this consists of multiple fields. So also if you renamed it to data it is in fact a message including a message. Sticking to the `message` as the object as it was already before, I would call the parameter `content` or  `body`.
What do you think? I mean if you think my argument is stupid then close this and
keep it like that.

Reply:
I changed the arg of `paymentMessage` from `message` to `subject`, so that I changed `data` to `message` and it is more coherent with other functions.